src/components/listing-components/login-dialog.tsx

```diff
@@ -35,12 +38,16 @@ const siteList: Site[] = [
  * Login Dialog component
  */
 const loginDialog: React.FC<LoginDialogProps> = ({ open, onClose, onLogin }) => {
-  const { surveyId } = useParams<{ surveyId: string }>();
   const navigate = useNavigate();
-  const [ site, setSite ] = useState(siteList[0].id);
-  const [ username, setUsername ] = useState("");
-  const [ password, setPassword ] = useState("");
-  const [ loginError, setLoginError ] = useState("");
+  const errorContext = React.useContext(ErrorContext);
+  const { surveyId } = useParams<{ surveyId: string }>();
+  const [ site, setSite ] = React.useState(siteList[0].id);
+  const [ username, setUsername ] = React.useState("");
+  const [ password, setPassword ] = React.useState("");
+  const [ loginError, setLoginError ] = React.useState("");
+  const [ accessToken, setAccessToken ] = React.useState("");
+  const [ refreshToken, setRefreshToken ] = React.useState("");
+  const [tokenExpiring, setTokenExpiring] = React.useState(false);
```

Comment:
That is true, I was thinking the opposed way because the token doesn't give a timestamp. So I changed it into 
[tokenTimestamp, setTokenTimestamp]
without it the code wouldn't know when to refresh